aten/src/ATen/native/native_functions.yaml

```diff
@@ -9524,15 +9524,19 @@
 
 - func: isposinf(Tensor self) -> Tensor
   variants: function, method
+  structured_delegate: isposinf.out
 
 - func: isposinf.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
+  structured: True
```

Comment:
Have you tried inheriting from `TensorIteratorBase`? You can do that with `structured_inherits: TensorIteratorBase`.

It looks like right now, you don't use TensorIteratorBase at all in the meta functions of `isneginf` and `isposinf`. Two reasons that would be useful are:
- We want to make sure that the meta function does all of the type promotion / broadcasting / handles other meta-like edge cases. If you use TensorIterator in the actual impl() function but not the meta() function, there's a risk of forgetting to do some checking in the meta function that's normally handled by TensorIterator.
- (More long-term) We eventually want to make all TensorIterator ops structured and use `TensorIteratorBase` everywhere, instead of `TensorIterator`.

Let me know if you run into issues with that - we've had some instances of it being hard to directly use `TensorIteratorBase`. I think you'll have to use `.declare_static_dtype_and_device(kBool, self.device())` in the config to force TensorIterator to treat the output tensor as a bool.

cc @ysiraichi in case you uncovered any other issues - I think you started looking into porting a few binary comparison ops.